overlord/ifacestate/handlers.go

```diff
@@ -744,6 +744,64 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// doDisconnectInterfaces creates tasks for disconnecting an interface and running its interface hooks.
+func (m *InterfaceManager) doDisconnectInterfaces(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapsup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapsup.Name()
+	connections, err := m.repo.Connections(snapName)
+	if err != nil {
+		return err
+	}
+
+	// check for conflicts on all connections first before creating disconnect hooks
+	for _, connRef := range connections {
+		ignore, err := findSymmetricInstallTask(st, connRef.PlugRef.Snap, connRef.SlotRef.Snap, task)
+		if err != nil {
+			return err
+		}
+
+		if ignore {
+			continue
+		}
+
+		if err := checkConnectConflicts(st, task.Change(), connRef.PlugRef.Snap, connRef.SlotRef.Snap, task); err != nil {
+			if _, retry := err.(*state.Retry); retry {
+				task.Logf("disconnect-interfaces task for snap %q will be retried because of %q - %q conflict", snapName, connRef.PlugRef.Snap, connRef.SlotRef.Snap)
+				return err // will retry
+			}
+			return fmt.Errorf("disconnect-interfaces conflict check failed: %s", err)
+		}
+	}
+
+	connectts := state.NewTaskSet()
```

Comment:
At first I thought this says `connections`. Not sure what's the better word but it's IMO too close and confusable.